codegen/scala-gen/src/main/scala/com/akkaserverless/codegen/scalasdk/impl/MainSourceGenerator.scala

```diff
@@ -0,0 +1,248 @@
+    }
+
+    val serviceImports = model.services.values.flatMap { serv =>
+      if (serv.fqn.parent.javaPackage != mainClassPackageName) {
+        serv match {
+          case actionServ: ModelBuilder.ActionService =>
+            List(actionServ.classNameQualified, actionServ.providerNameQualified)
+          case view: ModelBuilder.ViewService =>
+            List(view.classNameQualified, view.providerNameQualified)
+          case _ => Nil
+        }
+      } else List.empty
+    }
+
+    val otherImports = Nil
+    // FIXME
+//      model.services.values.flatMap { serv =>
+//      val types = serv.commands.flatMap { cmd =>
+//        cmd.inputType :: cmd.outputType :: Nil
+//      }
+//      collectRelevantTypes(types, serv.fqn).map { typ =>
+//        s"${typ.parent.javaPackage}.${typ.parent.javaOuterClassname}"
+//      }
+//    }
+
+    val entityContextImports = model.entities.values.collect {
+      case _: ModelBuilder.EventSourcedEntity =>
+        List("com.akkaserverless.scalasdk.eventsourcedentity.EventSourcedEntityContext")
+      case _: ModelBuilder.ValueEntity =>
+        List("com.akkaserverless.scalasdk.valueentity.ValueEntityContext")
+      case _: ModelBuilder.ReplicatedEntity =>
+        List("com.akkaserverless.scalasdk.replicatedentity.ReplicatedEntityContext")
+    }.flatten
+
+    val serviceContextImports = model.services.values.collect {
+      case _: ModelBuilder.ActionService =>
+        List("com.akkaserverless.scalasdk.action.ActionCreationContext")
+      case _: ModelBuilder.ViewService =>
+        List("com.akkaserverless.scalasdk.view.ViewCreationContext")
+    }.flatten
+    val contextImports = (entityContextImports ++ serviceContextImports).toSet
+
+    val entityCreators =
+      model.entities.values.toList
+        .sortBy(_.fqn.name)
+        .collect {
+          case entity: ModelBuilder.EventSourcedEntity =>
+            s"create${entity.fqn.name}: Function[EventSourcedEntityContext, ${entity.fqn.name}]"
+          case entity: ModelBuilder.ValueEntity =>
+            s"create${entity.fqn.name}: Function[ValueEntityContext, ${entity.fqn.name}]"
+          case entity: ModelBuilder.ReplicatedEntity =>
+            s"create${entity.fqn.name}: Function[ReplicatedEntityContext, ${entity.fqn.name}]"
+        }
+
+    val serviceCreators = model.services.values.toList
+      .sortBy(_.fqn.name)
+      .collect {
+        case service: ModelBuilder.ActionService =>
+          s"create${service.className}: Function[ActionCreationContext, ${service.className}]"
+        case view: ModelBuilder.ViewService =>
+          s"create${view.className}: Function[ViewCreationContext, ${view.className}]"
+      }
+
+    val creatorParameters = entityCreators ::: serviceCreators
+    val imports =
+      (List(
+        "com.akkaserverless.scalasdk.AkkaServerless") ++ entityImports ++ serviceImports ++ otherImports ++ contextImports).distinct.sorted
+        .map(pkg => s"import $pkg")
+        .mkString("\n")
```

Comment:
(could use `generateImports` here)

Reply:
thanks, done in 94cccf7a5e725516a27f4eb7e150e6949bb8a9f9